src/Flex.php

```diff
@@ -267,19 +191,12 @@ class_exists(__NAMESPACE__.str_replace('/', '\\', substr($file, \strlen(__DIR__)
                 BasePackage::$stabilities['dev'] = 1 + BasePackage::STABILITY_STABLE;
             }
 
-            if ($populateRepoCacheDir && isset(self::$repoReadingCommands[$command]) && ('install' !== $command || (file_exists($composerFile = Factory::getComposerFile()) && !file_exists(substr($composerFile, 0, -4).'lock')))) {
-                $this->populateRepoCacheDir();
-            }
-
             $app->add(new Command\RequireCommand($resolver, \Closure::fromCallable([$this, 'updateComposerLock'])));
             $app->add(new Command\UpdateCommand($resolver));
```

Comment:
Should those registration be changed to use the CommandProvider of the Composer 1.1+ plugin API instead of relying on the backtrace hack to access the Application ?

Reply:
I just had a look and I think there are little benefits in doing so. Limiting merge conflicts with `1.x` looks more important to me right now.